### PR TITLE
Radiation Storm Rework

### PR DESCRIPTION
**Pull request by CampinKiller24**
Ports the way radiation storms work from Bay. Unshielded areas are now hit with radiation, instead of mobs receiving the damage directly. Geiger counters detect radiation during radstorms now, and radiation-resistant clothing protects the wearer. Radiation does take some time to dissipate after the storm ends (approx 15-30 seconds, depending on the level), so the announcement has been updated to reflect that. Also adds geiger counters to the Engi-Vend.

Radiation exposure is now more harmful, especially extreme exposure, though it's still easily treated with proper medical care.

### CODE REVIEW

**Comment by CampinKiller24** [comment]
!review

**Review by NonQueueingMatt — APPROVED** [review]

File: code/modules/events/radiation_storm.dm [hunk 1]
```diff
@@ -42,8 +42,36 @@
...
 	for(var/mob/living/C in living_mob_list)
-		C.apply_radiation_effects()
+		var/area/A = get_area(C)
+		if(!A)
+			continue
+		if(isNotStationLevel(A.z))
+			continue
+		if(A.flags & RAD_SHIELDED)
+			continue
+		if(istype(C,/mob/living/carbon/human))
```
Comment (Doxxmedearly) [inline code/modules/events/radiation_storm.dm:57]:
```suggestion
		if(ishuman(C))
```

File: code/modules/events/radiation_storm.dm [hunk 2]
```diff
@@ -42,8 +42,36 @@
...
+			var/mob/living/carbon/human/H = C
+			
+			if(H.is_diona())
+				var/damage = rand(15, 30)
+				H.adjustToxLoss(-damage)
+				if(prob(5))
+					damage = rand(20, 60)
+					H.adjustToxLoss(-damage)
+				to_chat(H, SPAN_NOTICE("You can feel flow of energy which makes you regenerate."))
+
```
Comment (Doxxmedearly) [inline code/modules/events/radiation_storm.dm:67]: If we don't `continue` here, Dionae will get the mutation check that comes after. I don't know if they should get mutation chances? Might be a lore thing but unsure if a species that gets healed by radiation should potentially have their DNA be mutated by it.

File: code/modules/events/radiation_storm.dm [hunk 3]
```diff
@@ -42,8 +42,36 @@
...
+					H.adjustToxLoss(-damage)
+				to_chat(H, SPAN_NOTICE("You can feel flow of energy which makes you regenerate."))
+			
+			if(prob(5 * (1 - H.get_blocked_ratio(null, DAMAGE_RADIATION, damage_flags = DAMAGE_FLAG_DISPERSED, armor_pen = radiation_level))))
+				if (prob(75))
+					randmutb(H) // Applies bad mutation
+					domutcheck(H,null,MUTCHK_FORCED)
+				else
+					randmutg(H) // Applies good mutation
+					domutcheck(H,null,MUTCHK_FORCED)
```
Comment (Arrow768) [inline code/modules/events/radiation_storm.dm:74]: Thats some special snowflake code for a single event that should be generalized or removed.

-> The diona stuff is already handled in `/mob/living/carbon/proc/diona_handle_radiation(var/datum/dionastats/DS)`
-> The mutation stuff should be purged as its generally rather lrp (if we want more severe radiation effects then `/mob/living/carbon/human/handle_mutations_and_radiation()` should be used.

You could consider increasing the radiation strength to compensate for the lack of "additional action" in this proc.

File: code/modules/events/radiation_storm.dm (hunk 1 above)
Comment (Arrow768) [reply code/modules/events/radiation_storm.dm:57]: Irrelevant due to the comment below

File: code/modules/events/radiation_storm.dm (hunk 2 above)
Comment (Arrow768) [reply code/modules/events/radiation_storm.dm:67]: Irrelevant due to the comment below

**Review by Arrow768 — CHANGES_REQUESTED** [review]
In addition the proc `/proc/high_radiation_event()` should also be removed

File: code/modules/events/radiation_storm.dm (hunk 3 above)
Comment (CampinKiller24) [reply code/modules/events/radiation_storm.dm]: Made the above change to TRUE. Also nuked the snowflake code/mutation code, made the event output higher radiation levels, and increased the strength of radiation, especially at higher radiation levels. Extreme exposure is deadlier, though still treatable with proper medical care

File: code/modules/events/radiation_storm.dm [hunk 4]
```diff
@@ -42,8 +42,36 @@
 		lights()
 
 /datum/event/radiation_storm/proc/radiate()
+	var/radiation_level = rand(15, 30)
+	for(var/z in affecting_z)
+		SSradiation.z_radiate(locate(1, 1, z), radiation_level, 1)
```
[the remark this answers is not in the document]
Comment (CampinKiller24) [reply code/modules/events/radiation_storm.dm]: done

**Comment by CampinKiller24** [comment]
!review

**Review by Arrow768 — APPROVED** [review]

**Pull request merged**